File: test backend/backend/app/routers/mb_products.py
from fastapi import APIRouter, HTTPException
from app.services.mb_products_service import (
    list_motherboards,
    get_mb_details,
)

router = APIRouter(prefix="/api/products/mb", tags=["Motherboards"])

@router.get("")
def get_motherboards(
    socket: str | None = None,
    memory_type: str | None = None,
    form_factor: str | None = None,
    min_price: int | None = None,
    max_price: int | None = None,
    sort: str = "price_asc",
    limit: int = 24,
    offset: int = 0,
):
    filters = {
        "socket": socket,
        "memory_type": memory_type,
        "form_factor": form_factor,
        "min_price": min_price,
        "max_price": max_price,
    }

    return list_motherboards(filters, sort, limit, offset)


@router.get("/{product_id}")
def get_mb(product_id: int):
    data = get_mb_details(product_id)
    if not data:
        raise HTTPException(404, "Motherboard not found")
    return data

File: frontend/src/api/mbProducts.js
import { apiFetch } from "./api";

export function fetchMotherboards({
  search,
  sort = "price_asc",
  limit = 24,
  offset = 0,
  socket,
  memory_type,
  form_factor,
  min_price,
  max_price,
} = {}) {
  const params = new URLSearchParams();

  if (search) params.append("search", search);
  params.append("sort", sort);
  params.append("limit", limit);
  params.append("offset", offset);

  if (socket) params.append("socket", socket);
  if (memory_type) params.append("memory_type", memory_type);
  if (form_factor) params.append("form_factor", form_factor);
  if (min_price) params.append("min_price", min_price);
  if (max_price) params.append("max_price", max_price);

  return apiFetch(`/api/products/mb?${params.toString()}`);
}

export function fetchMbDetails(productId) {
  return apiFetch(`/api/products/mb/${productId}`);
}
